src/PrestaShopBundle/Resources/config/services/bundle/new_api.yml

```diff
@@ -54,3 +54,5 @@ services:
       - '@Symfony\Bridge\PsrHttpMessage\Factory\PsrHttpFactory'
       - '@Symfony\Bridge\PsrHttpMessage\Factory\HttpFoundationFactory'
       - '@Nyholm\Psr7\Factory\Psr17Factory'
+    calls:
+      - [ setContainer, [ '@service_container' ] ]
```

Comment:
Why is this needed exactly? 🤔 That's the kind of thing @FabienPapet is trying to remove on our other controllers because it's a bad practice and it prevents container optimization by Symfony framework (via private container for each controller)

I guess this controller doesn't extend the `FrameworkBundleAdminController` that [we use for detection to force this call in the compiler pass](https://github.com/PrestaShop/PrestaShop/blob/3d06748975e00c548e940bf82862d9d211ae94d4/src/PrestaShopBundle/DependencyInjection/Compiler/ContainerInjectionPass.php#L52), but if we can avoid relying on it it would be better Is it to access other services from the controller itself? Because all the services it's supposed to need are already injected in the constructor no? I don't know if using `getSubscribedServices` instead would solve the issue

Reply:
@jolelievre I think you're the best placed to tell us why is this needed since you've added this code on your own :)